.github/workflows/test.yml

```
@@ -90,36 +91,69 @@ jobs:
           filters: "tests/config/pytest_modules.yml"
           token: ""
 
+      - name: Fetch module tags
+        id: tags
+        run: |
+          echo modules=$(echo '${{ steps.filter.outputs.changes }}' | jq -c '. | map(select(contains("modules"))) | map(gsub("modules/"; ""))') >> $GITHUB_OUTPUT
+          echo subworkflows=$(echo '${{ steps.filter.outputs.changes }}' | jq '. | map(select(contains("subworkflow"))) | map(gsub("subworkflows/"; ""))') >> $GITHUB_OUTPUT
+
+      - name: debug
+        run: |
+          echo ${{ steps.tags.outputs.modules }}
+          echo ${{ steps.tags.outputs.subworkflows }}
+
   nf-test-changes:
     name: nf-test-changes
     runs-on: ubuntu-latest
     outputs:
-      # Expose matched filters as job 'modules' output variable
-      modules: ${{ steps.filter.outputs.changes }}
+      # Expose detected tags as 'modules' and 'workflows' output variables
+      paths: ${{ steps.list.outputs.changes }}
+      modules: ${{ steps.outputs.outputs.modules }}
+      subworkflows: ${{ steps.outputs.outputs.subworkflows}}
+
     steps:
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          architecture: "x64"
+
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - uses: actions/checkout@v3
         with:
-          fetch-depth: 2 # To retrieve the preceding commit.
+          fetch-depth: 0
 
-      - name: Combine all tags.yml files
-        id: get_tags
-        run: find . -name "tags.yml" -not -path "./.github/*" -exec cat {} + > .github/tags.yml
+      - name: Install gitpython find changed files
+        run: |
+          python -m pip install --upgrade pip
+          pip install gitpython pyyaml
 
-      - name: debug
-        run: cat .github/tags.yml
+      - name: List nf-test files
+        id: list
+        run: |
+          echo changes=$(python \
+            .github/python/find_changed_files.py \
+            -t pipeline workflow process \
+            --head_ref ${{ github.sha }} \
+            --base_ref origin/${{ github.base_ref }} \
+          ) >> $GITHUB_OUTPUT
+
+      - name: Separate modules and subworkflows
+        id: outputs
```

```suggestion
        id: components
```
find it less confusing than `outputs.outputs` and is common lingo in tools 🙂